app/policies/project_policy.rb

```diff
@@ -2,174 +2,209 @@
-    return true if Member.can_modify?(user, record.namespace) == true
+    return true if Member::AccessLevel.manageable.include?(effective_access_level)
 
     details[:name] = record.name
     false
   end
 
   def sample_listing?
-    return true if record.namespace.parent.user_namespace? && record.namespace.parent.owner == user
-    return true if Member.can_view?(user, record.namespace) == true
+    if effective_access_level > Member::AccessLevel::NO_ACCESS &&
+       effective_access_level != Member::AccessLevel::UPLOADER
+      return true
+    end
 
     details[:name] = record.name
     false
   end
 
   def create_sample?
-    return true if record.namespace.parent.user_namespace? && record.namespace.parent.owner == user
-    return true if Member.can_create_sample?(user, record.namespace) == true
+    return true if Member::AccessLevel.manageable.include?(effective_access_level)
+    return true if token_active(effective_access_level) == true
 
     details[:name] = record.name
     false
   end
 
   def destroy_sample?
-    return true if record.namespace.parent.user_namespace? && record.namespace.parent.owner == user
-    return true if Member.namespace_owners_include_user?(user, record.namespace) == true
+    return true if effective_access_level == Member::AccessLevel::OWNER
 
     details[:name] = record.name
     false
   end
 
   def read_sample?
-    return true if record.namespace.parent.user_namespace? && record.namespace.parent.owner == user
-    return true if Member.can_view?(user, record.namespace) == true
+    if effective_access_level > Member::AccessLevel::NO_ACCESS &&
+       effective_access_level != Member::AccessLevel::UPLOADER
+      return true
+    end
+    return true if token_active(effective_access_level) == true
 
     details[:name] = record.name
     false
   end
 
   def update_sample?
-    return true if record.namespace.parent.user_namespace? && record.namespace.parent.owner == user
-    return true if Member.can_modify_sample?(user, record.namespace) == true
+    return true if Member::AccessLevel.manageable.include?(effective_access_level)
+    return true if token_active(effective_access_level) == true
 
     details[:name] = record.name
     false
   end
 
   def transfer_sample?
-    return true if Member.can_transfer_sample?(user, record.namespace) == true
+    return true if Member::AccessLevel.manageable.include?(effective_access_level(false))
 
     details[:name] = record.name
     false
   end
 
   def transfer_sample_into_project?
-    return true if Member.user_has_namespace_maintainer_access?(user, record.namespace, false) &&
-                   Member.can_transfer_sample_to_project?(user, record.namespace, false) == true
-    return true if Member.can_transfer_sample_to_project?(user, record.namespace) == true
+    return true if effective_access_level(false) == Member::AccessLevel::MAINTAINER &&
```

Comment:
**nitpick (blocking):** Let's remove the passing of `false` here, for more detail see my comment on `effective_access_level` method definition.